src/test/java/com/google/devtools/build/lib/rules/cpp/CcCommonTest.java

```diff
@@ -1071,12 +1072,12 @@ public void testIncludeManglingSmoke() throws Exception {
     ConfiguredTarget lib = getConfiguredTarget("//third_party/a");
     CcCompilationContext ccCompilationContext = lib.get(CcInfo.PROVIDER).getCcCompilationContext();
     assertThat(ActionsTestUtil.prettyArtifactNames(ccCompilationContext.getDeclaredIncludeSrcs()))
-        .containsExactly("third_party/a/_virtual_includes/a/lib/b/c.h", "third_party/a/v1/b/c.h");
+        .containsExactly("_v_inc/207132b2/lib/b/c.h", "third_party/a/v1/b/c.h");
```

Comment:
Is this hash stable? Will we see test failures (and one-off updates) in otherwise unrelated CLs?

Reply:
Yes, the hash function is stable, input is basically the target path and name.

Reply:
Is there a way to only have this happen for Windows? I doubt it, but the loss in expressiveness of file paths is unfortunate.

Reply:
This would be a use case for `ctx.exec_platform_has_constraint`. But since `clang-cl` doesn't have this issue, it could be more appropriate to match on the `compiler` attribute of the toolchain or, even better, a new `hash_virtual_includes` feature.